src/react-apps/applications/runtime/src/components/base/DatepickerComponent.tsx

```diff
@@ -26,20 +26,28 @@ export class DatepickerComponent
     };
   }
 
-  public onDateChange = () => {
+  public onDateChange = (e: any) => {
+    this.setState({
+      value: e.target.value,
+    });
+  }
+
+  public ondateBlur = () => {
     setTimeout(() => {
-      if (!this.myDateCmp.current.value) {
+      if (!this.myDateCmp.current.value || this.state.value === this.myDateCmp.current.value) {
         return;
       }
       this.setState({
         value: this.myDateCmp.current.value,
       });
       this.props.handleDataChange(this.state.value);
-    }, 100);
+    }, 200);
   }
 
   public componentDidMount() {
-    (window as any).initDatePicker();
+    const dateFormat = 'dd.mm.yyyy';
+    const dateLanguage = 'no';
```

Comment:
Add todo-comment to take these values from props?

Reply:
fixed